### PR TITLE
Add Husky for pre-commit formatting hooks

### PR DESCRIPTION
**Pull request by queengooborg**
This PR adds `husky` as a dependency for pre-commit hooks to format all of the files before creating a commit.

### CODE REVIEW

**Comment by queengooborg** [comment]
That's the eventual plan!

File: .lintstagedrc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "*": "prettier --ignore-unknown --write",
+    "*.{js,jsx,tsx}": "eslint --fix",
+    "*.{json}": "npm run fix"
```
Comment (nschonni) [inline .lintstagedrc:4]: Prettier stuff will already be run on the wildcard
```suggestion
    "*.{json}": "ts-node scripts/fix/index.ts"
```

**Review by Elchi3 — COMMENTED** [review]
Let's ask in the next owner's meeting about pre-commit hooks. I'm not against it per se, just want to make sure the majority appreciates these ergonomics. Also, might be nice to mention this in the contribution docs if we introduce them.

**Comment by queengooborg** [comment]
MDN content just landed this, so let's land this here as well!

**Pull request merged**